### PR TITLE
discover/grub2: Trim fields after BLS grub variable expansion

### PR DESCRIPTION
**Pull request by rarbab**
Consider this line in a BLS config:
```
initrd /initramfs-5.14.0-70.13.1.el9_0.ppc64le.img $tuned_initrd
```
If `$tuned_initrd` is empty or unset, the initrd will be parsed as `"/initramfs-5.14.0-70.13.1.el9_0.ppc64le.img "`, with a trailing space. Since this file doesn't exist, trying to boot the entry will fail and report `"Error loading initrd"`.

Add a test case for this. To fix, trim leading and trailing blank space from fields after the grub variables have been expanded.

### CODE REVIEW

File: test/parser/test-grub2-blscfg-opts-grubenv.c [hunk 1]
```diff
@@ -21,7 +21,7 @@ void run_test(struct parser_test *test)
 			     "/loader/entries/6c063c8e48904f2684abde8eea303f41-4.15.2-302.fc28.x86_64.conf",
 			     "title Fedora (4.15.2-302.fc28.x86_64) 28 (Twenty Eight)\n"
 			     "linux /vmlinuz-4.15.2-302.fc28.x86_64\n"
-			     "initrd /initramfs-4.15.2-302.fc28.x86_64.img\n"
+			     "initrd $blank /initramfs-4.15.2-302.fc28.x86_64.img $blank\n"
```
Comment (jk-ozlabs) [inline test/parser/test-grub2-blscfg-opts-grubenv.c:24]: This might warrant its own test rather than adding into the env test here.

File: discover/grub2/blscfg.c [hunk 2]
```diff
@@ -55,6 +55,25 @@ static char *field_append(struct bls_state *state, int type, char *buffer,
...
 }
 
+static void field_trim(char *buffer)
+{
+        int i, j, end = 0;
+
+        if (!buffer)
+                return;
+
+        for (i = 0; isblank(buffer[i]); i++);
```
Comment (jk-ozlabs) [inline discover/grub2/blscfg.c:65]: Keep in mind that `conf_get_pair_space` will be splitting on spaces only, while this will be stripping tabs and spaces. Is that OK?

**Review by jk-ozlabs — COMMENTED** [review]
Looks good to me, thanks for the change. A couple of comments inline, but I'm okay with merging this as-is if you're happy with things.

File: test/parser/test-grub2-blscfg-opts-grubenv.c (hunk 1 above)
Comment (rarbab) [reply test/parser/test-grub2-blscfg-opts-grubenv.c:24]: That's true, it could be spun into a whole new test, but I think of it as an edge case of what's already being tested there with `$kernelopts`; variable expansion, but a case where the variable is blank.

File: discover/grub2/blscfg.c (hunk 2 above)
Comment (rarbab) [reply discover/grub2/blscfg.c:65]: Yes, that's fine. The issue I've described would still be resolved if I just checked `buffer[i] == ' '` instead of `isblank(buffer[i])`, but there's no harm in stripping any errant tabs as well. That was just me trying to make it a little more robust.

**Comment by rarbab** [comment]
Thanks for reviewing! Yes, I'm okay with merging as-is.
